src/modm/math/utils/misc.hpp

```diff
@@ -56,74 +57,58 @@ pow(uint32_t base, uint8_t exponent)
 }
 
 /**
- * This does what you think it does.
+ * @brief 		Variadic min for 2-∞ objects
+ *
+ * @param  a 	first object to compare
+ * @param  b  	second object to compare
+ * @param  cs  	Further objects for comparison
  *
- * @param  a  A thing of arbitrary type.
- * @param  b  Another thing of arbitrary type.
- * @return   The lesser of the parameters.
+ * @return   	The smallest object
  *
- * This is the simple classic generic implementation.  It will work on
- * temporary expressions, since they are only evaluated once, unlike a
- * preprocessor macro.
+ * @see			https://stackoverflow.com/questions/23815138/implementing-variadic-min-max-functions
  */
 template<typename T>
-inline const T&
-min(const T& a, const T& b)
+constexpr T vmin(T a, T b)
 {
-	if (b < a)
-		return b;
-	else
-		return a;
+    return a < b ? a : b;
 }
 
-/**
- * This does what you think it does.
- *
- * @param  a  A thing of arbitrary type.
- * @param  b  Another thing of arbitrary type.
- * @return   The greater of the parameters.
- *
- * This is the simple classic generic implementation.  It will work on
- * temporary expressions, since they are only evaluated once, unlike a
- * preprocessor macro.
- */
-template<typename T>
-inline const T&
-max(const T& a, const T& b)
+template<typename T, typename... Ts>
+constexpr T vmin(T a, T b, Ts&&... cs)
 {
-	if (a < b)
-		return b;
-	else
-		return a;
+    return a < b ? vmin(a, cs...) : vmin(b, cs...);
 }
 
 /**
  * This does what you think it does.
  *
- * @param  a  A thing of arbitrary type.
- * @param  b  Another thing of arbitrary type.
- * @param  c  Something else of arbitrary type.
- * @return   The greater of the three parameters.
+ * @param  a 	first object to compare
+ * @param  b  	second object to compare
+ * @param  cs  	Further objects to compare
  *
- * This is the simple classic generic implementation.  It will work on
- * temporary expressions, since they are only evaluated once, unlike a
- * preprocessor macro.
+ * @return   	The biggest object
+ *
+ * @see			https://stackoverflow.com/questions/23815138/implementing-variadic-min-max-functions
  */
 template<typename T>
-constexpr T
-max(const T a, const T b, const T c)
+constexpr T vmax(T& a, T& b)
```

Comment:
```suggestion
constexpr T vmax(const T& a, const T& b)
```
The present implementations does not accept constants, only non-const lvalues.